### PR TITLE
remove the roadmap from the sphinx based docs

### PR DESCRIPTION
**Pull request by esc**
As discovered in https://github.com/numba/numba/issues/7930 the roadmap
isn't being actively maintained. This may cause confusion and create the
impression that the project is no longer active. So for now, no
information is better than outdated

If any of these items are still relevant and need to be done they ought
to be converted to tickets.

### CODE REVIEW

File: docs/source/developer/roadmap.rst [hunk 1]
```diff
@@ -1,77 +0,0 @@
...
-* Improve memory management of Numba-allocated memory
-* Better support for writing code transformation passes
-* Make caching and parallel execution features opt-out instead of opt-in
-
-  * add heuristic to determine if parfor passes will be beneficial
-
-Long Term: 2020 and beyond
-==========================
-
-* Unify GPU backends (share more code and interfaces)
```
Comment (gmarkall) [inline docs/source/developer/roadmap.rst:62]: xrefs: #7814 #7815

File: docs/source/developer/roadmap.rst [hunk 2]
```diff
@@ -1,77 +0,0 @@
...
-* Improve ahead of time compilation (for low powered devices)
-* Improve cross language connections (C++, JVM?, Julia?, R?)
-
-  * Call Numba from other languages,
-  * Call from Numba into other languages
```
Comment (gmarkall) [inline docs/source/developer/roadmap.rst:67]: xref: #7860

File: docs/source/developer/roadmap.rst [hunk 3]
```diff
@@ -1,77 +0,0 @@
...
-
-* Better support for "hybrid" CPU/GPU/TPU/etc programming
```
Comment (gmarkall) [inline docs/source/developer/roadmap.rst:69]: xrefs: #6944

File: docs/source/developer/roadmap.rst [hunk 4]
```diff
@@ -1,77 +0,0 @@
...
-* Partial / deferred compilation of functions
-* Foster integration of Numba into core PyData packages:
-
-  * scipy/scikit-learn/scikit-image/pandas
-
-* More support for efforts to put Numba into other applications (databases, etc) for compiling user-defined functions
```
Comment (gmarkall) [inline docs/source/developer/roadmap.rst:75]: xrefs: https://github.com/gmarkall/numba-accelerated-udfs

File: docs/source/developer/roadmap.rst [hunk 5]
```diff
@@ -1,77 +0,0 @@
...
-Medium Term: 2019H2
-===================
-
-* Unify dispatch of regular functions, ufuncs, and gufuncs
-* Declare Numba 1.0 with stable interfaces
-* Continue to improve usability and debugging (see above)
-* Continue refactoring Numba internals to solve common bug reports (see above)
-* JIT class review and improvement
-* Improve compilation speed
-* Improve memory management of Numba-allocated memory
```
Comment (gmarkall) [inline docs/source/developer/roadmap.rst:53]: xref: #5328

**Comment by gmarkall** [comment]
This looks good to me. I've added some links for PRs that worked towards some of the roadmap items for CUDA, in case anyone viewing the old roadmap in this PR is interested in those already-completed items.

**Review by sklam — APPROVED** [review]

**Pull request merged**